cmd/cmd.js

```diff
@@ -180,19 +180,21 @@ class Cmd {
       .action(function(env, options) {
         checkDeps();
         i18n.setOrDetectLocale(options.locale);
-        const nullify = (v) => (!v || typeof v !== 'string') ? null : v;
+        const nullify = (v) => {
+          return (!v || typeof v !== 'string') ? null : v;
+        };
         embark.run({
           env: env || 'development',
           serverPort: nullify(options.port),
           serverHost: nullify(options.host),
           client: options.client || 'geth',
           locale: options.locale,
-          runWebserver: options.noserver == null ? null : !options.noserver,
+          runWebserver: !options.noserver ? null : false,
```

Comment:
I don't think this can work either. `cmd_controller.js` is expecting `null` as a signal value that a cli option was not specified. If a cli option is specified, it will override a setting in `config/webserver.js` during the recursive merge in core/config. If no cli option was specified we don't want a `webServerConfig.<property>` to be written so it doesn't clobber the value coming from `config/webserver.js`. This is the case for `noserver` (which becomes `runWebserver`), `nobrowser` (which becomes `openBrowser`), `port`, and `host`. For `port` and `host` the nullify function is satisfactory, but because of how no/server/browser get flipped around, I coded them specially.

If we're not using `== null` then your first instinct was right, should be:

```
runWebserver: options.noserver === null || options.noserver === undefined ? null : !options.noserver,
```

This is a tricky one, and I made the changes I did when realizing that the way we had it before, the config option was getting clobbered by the cli option whether specified or not. I think we do want the config option to be honored unless the cli option is used to override it.

Reply:
Actually, this should be okay:
```
runWebserver: options.noserver === undefined ? null : !options.noserver,
```
If `--noserver` isn't used, `options.noserver` will be `undefined`.

Could also change to:
```
runWebserver: options.noserver === undefined ? undefined : !options.noserver,
```
BUT in that case, in `cmd_controller.js` the `if` blocks for runWebserver, openBrowser, port, and host should be changed to test for `=== undefined`.

Reply:
I mean, I just tested with what's above, and it works great with or without the flag. Without flag, `noServer` is falsy, it returns `null`. With the flag, `noServer` is true, so `!noServer` would be false, same as above.

Reply:
```
runWebserver: !options.noserver ? null : false,
```

Maybe it's Friday and so I'm not interpreting that line correctly, but here goes:

If `--noserver` is *not* specified on the command-line then `options.noserver` will be `undefined` and therefore `!options.noserver` will be `true`, which will set `runWebserver` to `null`. Because `runWebserver === null` then `cmd_controller.js` will not set any `webServerConfig.enabled` property and the value in the DApp's `config/webserver.js` (or embark's default if not set in the config) will take effect. 👍 

If `--noserver` *is* specified on the command-line then `options.noserver` will be `false` and therefore `!options.noserver` will be `true`, which will set `runWebserver` to `null`. Because `runWebserver === null` then `cmd_controller.js` will not set any `webServerConfig.enabled` property and the value in the DApp's `config/webserver.js` (or embark's default if not set in the config) will take effect. In other words, `--noserver` will have no effect. 👎 

Compare with:

```
runWebserver: options.noserver === undefined ? null : !options.noserver,
```

If `--noserver` is *not* specified on the command-line then `options.noserver` will be `undefined` and therefore `options.noserver === undefined` will be `true`, which will set `runWebserver` to `null`. Because `runWebserver === null` then `cmd_controller.js` will not set any `webServerConfig.enabled` property and the value in the DApp's `config/webserver.js` (or embark's default if not set in the config) will take effect. 👍 

If `--noserver` *is* specified on the command-line then `options.noserver` will be `false` and therefore `options.noserver === undefined` will be `false`, which will set `runWebserver` to `true` (`!options.noserver`). Because `runWebserver !== null` then `cmd_controller.js` will set `webServerConfig.enabled = true`, overriding the value in the DApp's `config/webserver.js` (or embark's default if not set in the config). 👍

Reply:
> If --noserver is specified on the command-line then options.noserver will be false 
That's where you are wrong. If it is specified, `noserver` becomes true.

Reply:
You're totally right, I flipped around the `--noserver` in my head when reading your expression. When accounting for that, the two expressions are completely equivalent, accounting for the fact that `options.noserver === undefined` is false whether `options.noserver` is `true`/`false`.

Final point on this:

It might be better to do:
```
runWebserver: !options.noserver ? undefined : false,
```
And then change the tests in cmd_controller.js to check for `!== undefined`. That way, if some other code than cmd.js constructs EmbarkController it's okay if in the options passed to the constructor some options are just left out.